web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,280 @@
+            var output = [];  
+            Autocomplete.get({term: string},
+                function success(response) {
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.filterSearch = response;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );  
+        }  
+
+        $scope.fillTextbox = function(string){  
+            $scope.options.query = string;  
+            $scope.hidethis = true;  
+            $scope.hideoption = false;
+            document.getElementById("search-btn").focus();
+        }  
+
+        $scope.search = function(options) {
+            $rootScope.searchOptions = angular.copy(options);
+            $location.path("/metadataView/" + options.query + '/' + options.opt);
+        };
+
+        $scope.$watch(function () { return SearchOptions.getSearchOptions(); }, function (newValue, oldValue) {
+            if (newValue != null) {
+                $scope.options.query= newValue.query;
+                $scope.options.opt= newValue.opt;
+            }
+        }, true);
+    }]);
+    
+mudrodControllers.controller('vocabularyCtrl', ['$scope', '$rootScope', 'VocabList',
+    function vocabularyCtrl($scope, $rootScope, VocabList) {
+
+        var word = $rootScope.searchOptions.query;
+        var opt = $rootScope.searchOptions.opt;
+        VocabList.get({query: word},
+                function success(response) {
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.ontologyList = response.graph.ontology;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+        );
+    }]);
+
+mudrodControllers.controller('metadataViewCtrl', ['$rootScope', '$scope', '$routeParams', 'MetaData', 'PagerService', 'SearchOptions', 
+    function metadataViewCtrl($rootScope, $scope, $routeParams, MetaData, PagerService, SearchOptions) {
+        var vm = this;
+        vm.PDItems = [];
+        vm.pager = {};
+        vm.setPage = setPage;
+        vm.rankData = rankData;
+        vm.totalMatches = 0;
+        
+
+        var word = new String();
+        var opt = new String();
+        var rankopt = 'Rank-SVM';
+        
+        if(!$routeParams.query){
+            word = $rootScope.searchOptions.query;    
+            opt = $rootScope.searchOptions.opt; 
+            SearchOptions.setSearchOptions({'query':word, 'opt':opt});
+        } else {
+            word = $routeParams.query;
+            opt = $routeParams.opt;
+            
+            var searchKeyWords = new String();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a literal value for this. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3APrimitiveWrappers)